extensions/parquet/table/src/main/java/io/deephaven/parquet/table/pagestore/topage/ToInstantPageFromInt96.java

```diff
@@ -91,11 +93,18 @@ public final long[] convertResult(@NotNull final Object result) {
         final long[] resultLongs = new long[resultLength];
 
         for (int ri = 0; ri < resultLength; ++ri) {
-            final ByteBuffer resultBuffer = ByteBuffer.wrap(results[ri].getBytesUnsafe());
-            resultBuffer.order(java.nio.ByteOrder.LITTLE_ENDIAN);
-            final long nanos = resultBuffer.getLong();
-            final int julianDate = resultBuffer.getInt();
-            resultLongs[ri] = (julianDate - JULIAN_OFFSET_TO_UNIX_EPOCH_DAYS) * (NANOS_PER_DAY) + nanos + offset;
+            if (results[ri] != null) {
+                final ByteBuffer resultBuffer = ByteBuffer.wrap(results[ri].getBytesUnsafe());
+                resultBuffer.order(java.nio.ByteOrder.LITTLE_ENDIAN);
+                final long nanos = resultBuffer.getLong();
+                final int julianDate = resultBuffer.getInt();
+                resultLongs[ri] = (julianDate - JULIAN_OFFSET_TO_UNIX_EPOCH_DAYS) * (NANOS_PER_DAY) + nanos + offset;
+            }
+            else {
+                // TODO What is the difference between NULL_LONG and BOXED?
+                // BOXED version is used for int64 instants, so using the same here
+                resultLongs[ri] = NULL_LONG_BOXED;
```

Comment:
`NULL_LONG_BOXED` is a boxed type (`java.lang.Long`); `NULL_LONG` is a primitive type (`long`). So you should use `NULL_LONG_BOXED` when you want to store a null long as a reference type (e.g., `Object`/`Number`/`Long`) and `NULL_LONG` when you want to store a null long as a primitive `long`.

It looks to me like you should use `NULL_LONG` here; `resultLongs[ri]` is a `long` so the `NULL_LONG_BOXED` is going to get [unboxed](https://docs.oracle.com/javase/tutorial/java/data/autoboxing.html) anyway.

Reply:
Thanks for explaining Raffi, I will make the change.

Reply:
Thanks for expanding Ryan. It would not be possible to use `null` in this case because we are returning an array of primitive long type. Therefore, I have kept the behavior similar to the int64 based timestamps. 
Let me ask Christian to have a look since he wrote this code originally.